iothub/device/src/Transport/ErrorDelegatingHandler.cs

```diff
@@ -143,7 +144,8 @@ private static bool IsTlsSecurity(Exception singleException)
 
         private static bool IsNetwork(Exception singleException)
         {
-            return s_networkExceptions.Contains(singleException.GetType());
+            Type exceptionType = singleException.GetType();
+            return s_networkExceptions.Any(baseExceptionType => baseExceptionType.IsAssignableFrom(exceptionType));
```

Comment:
You can also do this - 
s_networkExceptions.Any(ne => ne.IsInstanceOfType(singleException))

Reply:
IsInstanceOfType will handle interface as well. Should I need to use it?